### PR TITLE
convert document translator swagger ->tsp

### PR DESCRIPTION
**Pull request by vikaspalaskar**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

### CODE REVIEW

File: specification/translation/Azure.AI.DocumentTranslation/routes.tsp [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+  @route("/batches")
+  @post
+  startTranslation is Azure.Core.Foundations.Operation<
+    StartTranslationDetails,
+    void
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Doesn't fit standard ops"
+  @summary("Returns a list of batch requests submitted and the status for each request")
+  @doc("""
```
Comment (allenjzhang) [inline specification/translation/Azure.AI.DocumentTranslation/routes.tsp:43]: For these really lengthy doc comment and summary, I suggest move to a separate file and use augment decorator style (@@doc) to keep the code easier to read.

File: specification/translation/Azure.AI.DocumentTranslation/routes.tsp [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Doesn't fit standard ops"
+  @summary("Returns a list of supported storage sources")
+  @doc("""
+Returns a list of storage sources/options supported by the Document Translation
+service.
+""")
+  @route("/storagesources")
+  @get
+  getSupportedStorageSources is Azure.Core.Foundations.Operation<
+    {},
```
Comment (allenjzhang) [inline specification/translation/Azure.AI.DocumentTranslation/routes.tsp:434]: void or empty {}?

File: specification/translation/Azure.AI.DocumentTranslation/main.tsp [hunk 3]
```diff
@@ -9,84 +10,33 @@ using TypeSpec.Versioning;
...
-    /** Supported document Translation endpoints (protocol and hostname, for example: https://{TranslatorResourceName}.cognitiveservices.azure.com/translator). */
-    Endpoint: url,
+    endpoint: string,
   }
 )
-@versioned(DocumentTranslation.Versions)
-namespace DocumentTranslation;
+@doc("// FIXME: (miissing-service-description) Add service description")
+@versioned(DocumentTranslationRestClient.Versions)
+namespace DocumentTranslationRestClient;
```
Comment (chunyu3) [inline specification/translation/Azure.AI.DocumentTranslation/main.tsp:35]: The namespace should be `DocumentTranslation`
```suggestion
namespace DocumentTranslation;
```

**Comment by raych1** [comment]
@weidongxu-microsoft , it seems the generated code of Java has code style issue based on the error log as follows?
```
cmderr	[generate.py] java.lang.RuntimeException: Failed to complete postprocessing.
cmderr	[generate.py] 	at com.azure.autorest.postprocessor.Postprocessor.postProcess(Postprocessor.java:50)
cmderr	[generate.py] 	at com.azure.typespec.Main.handleDPG(Main.java:143)
cmderr	[generate.py] 	at com.azure.typespec.Main.main(Main.java:90)
cmderr	[generate.py] Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: com.google.googlejavaformat.java.FormatterException: error: Expected ; after import
cmderr	[generate.py] 	at com.azure.autorest.postprocessor.Postprocessor.writeToFiles(Postprocessor.java:124)
cmderr	[generate.py] 	at com.azure.autorest.postprocessor.Postprocessor.postProcess(Postprocessor.java:47)
cmderr	[generate.py] 	... 2 more
```

File: specification/translation/Azure.AI.DocumentTranslation/tspconfig.yaml [hunk 4]
```diff
@@ -16,19 +16,19 @@ options:
...
-    package-dir: "azure-ai-translator-document"
-    package-name: "azure-ai-translator-document"
+    package-dir: "azure-ai-translation-document"
+    package-name: "azure-ai-translation-document"
     package-mode: dataplane
   "@azure-tools/typespec-java":
-    package-dir: "azure-ai-translator-document"
-    namespace: com.azure.ai.translator.document
+    package-dir: "azure-ai-translation-document"
+    namespace: azure-ai-translation-document
```
Comment (weidongxu-microsoft) [inline specification/translation/Azure.AI.DocumentTranslation/tspconfig.yaml:29]: Java namespace is wrong

**Comment by weidongxu-microsoft** [comment]
> @weidongxu-microsoft , it seems the generated code of Java has code style issue based on the error log as follows?
> 
> ```
> cmderr	[generate.py] java.lang.RuntimeException: Failed to complete postprocessing.
> cmderr	[generate.py] 	at com.azure.autorest.postprocessor.Postprocessor.postProcess(Postprocessor.java:50)
> cmderr	[generate.py] 	at com.azure.typespec.Main.handleDPG(Main.java:143)
> cmderr	[generate.py] 	at com.azure.typespec.Main.main(Main.java:90)
> cmderr	[generate.py] Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: com.google.googlejavaformat.java.FormatterException: error: Expected ; after import
> cmderr	[generate.py] 	at com.azure.autorest.postprocessor.Postprocessor.writeToFiles(Postprocessor.java:124)
> cmderr	[generate.py] 	at com.azure.autorest.postprocessor.Postprocessor.postProcess(Postprocessor.java:47)
> cmderr	[generate.py] 	... 2 more
> ```

https://github.com/Azure/azure-rest-api-specs/pull/27420#discussion_r1519236244

File: specification/translation/Azure.AI.DocumentTranslation/tspconfig.yaml [hunk 5]
```diff
@@ -16,19 +16,19 @@ options:
...
   "@azure-tools/typespec-ts":
-    package-dir: "azure-ai-translator-document"
+    package-dir: "azure-ai-translation-document"
```
Comment (MaryGao) [inline specification/translation/Azure.AI.DocumentTranslation/tspconfig.yaml:31]: Is the same package with `@azure-rest/ai-document-translator`?

https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/documenttranslator/ai-document-translator-rest

**Pull request closed without merging**